File: Lab1/Result.cs
using System;
namespace Lab1
{
	public class Result
	{
		int total_value;
		int total_weight { get; set; }
		List<int> items;

		public Result()
		{
			total_value = 0;
			total_weight = 0;
			items = new List<int>();

		}

		public void Add(Item item)
		{
			items.Add(item.GetId());
			total_value += item.GetValue();
			total_weight += item.GetWeight();
		}

		public List<int> GetItems() { return items; }
		public int GetTotalVal() { return total_value; }
		public int GetTotalWeight() { return total_weight; }

		public void PrintResultList()
		{
			Console.Write("IDs:");
			items.ForEach(Console.Write);
			Console.Write("\n");

			Console.WriteLine($"total_weight: {total_weight}");
			Console.WriteLine($"total_value: {total_value}");
		}

        public override string ToString()
        {
			String temp = new String("");
			foreach (int i in items) { temp += (i.ToString() + "\r\n"); };
			return temp;
        }
    }
}

File: Lab5/src/main/java/org/example/Result.java
package org.example;

import java.util.ArrayList;
import java.util.List;

public class Result {
    List<Item> items;
    int amount;
    int totalValue;
    int totalWeight;


    public Result(){
        items = new ArrayList<>();
        amount=0;
        totalValue=0;
        totalWeight=0;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("Result: \n");
        sb.append("Amount: ").append(amount).append("\n");
        sb.append("Total value: ").append(totalValue).append("\n");
        sb.append("Total weight: ").append(totalWeight).append("\n");
        sb.append("Selected items: \n");
        for(Item item : items) {
            sb.append(item).append("\n");
        }
        return sb.toString();
    }
}
